src/map/core.rs

```diff
@@ -183,7 +183,6 @@ impl<K, V> IndexMapCore<K, V> {
         }
     }
 
-    #[track_caller]
```

Comment:
Only your addition on the internal `insert_bulk_no_grow` should have been removed. All of these other existing internal instances are needed to track public API panics. In other words, your overall PR should not show any removals, only the agreed additions.

Once you've done that, please squash your fixup commits as well. The closure change can be kept separately if you like, or it would be fine to squash that down too.

Reply:
Note: while I only commented on this one, **everything** that existed before your PR should be restored.